Bottom: 50dab70ad7eb3dbdef71c75d4d93c9cb91e57f58
Top:    df78a11fcb050d8c6c85e5688e1fd4109f8ae8d7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 17:47:37 +0000

polygon1.c: Fix unitialised output polygon upon when hitting allocation failure

This was a bug present in the poly_Boolean() routine. We need to set the output
polygon to NULL if we are going to return early on memory failure. This was
noticed during general review after observing problems due to our copy routines
incorrectly returning FALSE when copying a NULL polygon.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 1dd9607..f970086 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2400,6 +2400,8 @@ poly_Boolean (const POLYAREA * a_org, const POLYAREA * b_org,
 {
   POLYAREA *a = NULL, *b = NULL;
 
+  *res = NULL;
+
   if (!poly_M_Copy0 (&a, a_org) || !poly_M_Copy0 (&b, b_org))
     return err_no_memory;
